src/app/(employee)/getting-started/_components/CertificateModel.tsx

```diff
@@ -1,59 +1,167 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
 import { IMAGES } from '@/assets';
 import Image from 'next/image';
 import Input from "@/components/Input";
 import Button from '@/components/Button';
 
-const CertificateModel = () => {
-    return (
-        <div>{/* The button to open modal */}
-            <div className=' flex justify-end'>
-                <label htmlFor="my_modal_7" className=" bg-white">
-                    <div className=' flex justify-end gap-2 cursor-pointer'>
-                        <span className=' text-primary-500 text-[16px] font-600'>Add More</span>
-                        <Image src={IMAGES.circle} alt='circle' />
-                    </div>
-                </label>
+interface Certification {
+  name: string;
+  issuingOrganization: string;
+  issueDate: string;
+  expirationDate: string;
+  credentialID: string;
+  credentialURL: string;
+}
+
+interface CertificateModelProps {
+  addCertification: (certification: Certification) => void;
+  showOnMount: boolean;
+}
+
+const CertificateModel: React.FC<CertificateModelProps> = ({ addCertification, showOnMount }) => {
+  const [certification, setCertification] = useState<Certification>({
+    name: '',
+    issuingOrganization: '',
+    issueDate: '',
+    expirationDate: '',
+    credentialID: '',
+    credentialURL: '',
+  });
+
+  const [validationError, setValidationError] = useState<string>('');
+
+  useEffect(() => {
+    if (showOnMount) {
+      const modalElement = document.getElementById('my_modal_7') as HTMLInputElement | null;
+      if (modalElement) {
+        modalElement.checked = true;
+      }
+    }
+  }, [showOnMount]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const { id, value } = e.target;
+    setCertification((prev) => ({ ...prev, [id]: value }));
+  };
+
+  const handleAdd = () => {
+    // Simple validation
+    if (!certification.name || !certification.issuingOrganization || !certification.issueDate || !certification.expirationDate) {
+      setValidationError('Please fill out all required fields.');
+      return;
+    }
+
+    addCertification(certification);
+    setCertification({
+      name: '',
+      issuingOrganization: '',
+      issueDate: '',
+      expirationDate: '',
+      credentialID: '',
+      credentialURL: '',
+    });
+
+    // Clear validation error
+    setValidationError('');
+
+    // Close modal after adding
+    const modalElement = document.getElementById('my_modal_7') as HTMLInputElement | null;
+    if (modalElement) {
+      modalElement.checked = false;
+    }
+  };
+
+  return (
+    <div>
+      {/* Modal Trigger Button */}
+      <div className='flex justify-end'>
+        <label htmlFor="my_modal_7" className="bg-white">
```

Comment:
The id of the modal must be unique for each modal and meaningful because daisyui modal depends upon id so give each modal unique and meaningful id not like my_modal_7